### PR TITLE
Expand gemma_rmsnorm input dimension

### PR DESCRIPTION
**Pull request by MeowMiaoJ**
According to [gemma3 model in sglang](https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/models/gemma3_causal.py), `gemma_rmsnorm` may receive a 4d input. This PR expand the input dimension to flattenable/unflattenable 4d shape and add corresponding test.

### CODE REVIEW

File: src/sycl/RMSNorm.cpp [hunk 1]
```diff
@@ -48,15 +48,25 @@ struct RowStrides {
...
-  int64_t outer_stride = t.stride(0);
-  int64_t inner_size = t.size(1);
-  int64_t inner_stride = t.stride(1);
-  if (t.size(0) == 1 || outer_stride == inner_size * inner_stride) {
+  if (t.dim() == 4) {
+    // 4D only: the leading batch-like dimension (dim 0) must be size 1,
+    // since our two-level (outer, inner) stride formula cannot represent
+    // a third level of striding.
+    TORCH_CHECK(
+        t.size(0) == 1, "get_row_strides: leading dimension 0 must have size 1 for a 4D tensor, got size ", t.size(0));
```
Comment (CaoE) [inline src/sycl/RMSNorm.cpp]: Can we support cases of `t.size(0) > 1` ?

File: src/sycl/RMSNorm.cpp [hunk 2]
```diff
@@ -48,15 +48,25 @@ struct RowStrides {
...
+  }
+
+  // For 3D/4D tensors, the outer is the second-to-last dimension and
+  // the inner is the last dimension.
+  int64_t outer_stride = t.stride(-3);
+  int64_t inner_size = t.size(-2);
+  int64_t inner_stride = t.stride(-2);
+  if (t.size(-3) == 1 || outer_stride == inner_size * inner_stride) {
```
Comment (CaoE) [inline src/sycl/RMSNorm.cpp]: Please be aware of these copilot comments and ensure the tests cover these corner cases.

File: src/sycl/RMSNorm.cpp (hunk 1 above)
Comment (MeowMiaoJ) [reply src/sycl/RMSNorm.cpp]: We don't support `t.size(0) > 1` because [the only 4d size of gemma3's q/k input is [1, h, s, head_dim]](https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/models/gemma3_causal.py#L287-L295). We can support `t.size(0) > 1` for extensibility.

File: src/sycl/RMSNorm.cpp (hunk 2 above)
Comment (MeowMiaoJ) [reply src/sycl/RMSNorm.cpp]: Cover tests by [new commit](https://github.com/sgl-project/sgl-kernel-xpu/pull/367/changes/6043b0b9ee66a35d88aae32d068f66a7d4228ccc).

File: src/sycl/RMSNorm.cpp (hunk 1 above)
Comment (CaoE) [reply src/sycl/RMSNorm.cpp]: From the perspective of the OP's functionality, we should support full 4D inputs, and I think this shouldn't require much modification.

File: src/sycl/RMSNorm.cpp [hunk 3]
```diff
@@ -48,15 +48,25 @@ struct RowStrides {
...
+
+  // For 3D/4D tensors, the outer is the second-to-last dimension and
+  // the inner is the last dimension.
+  int64_t outer_stride = t.stride(-3);
+  int64_t inner_size = t.size(-2);
+  int64_t inner_stride = t.stride(-2);
+  if (((t.dim() == 3) && (t.size(0) == 1)) || t.size(-2) == 1 || outer_stride == inner_size * inner_stride) {
     // Flattenable: a single stride describes all rows.
     return {inner_stride, 1, 0};
   }
```
[the remark this answers is not in the document]
Comment (CaoE) [reply src/sycl/RMSNorm.cpp]: Please note the 3D/4D input with corner cases where size=1.

[the remark this answers is not in the document]
Comment (MeowMiaoJ) [reply src/sycl/RMSNorm.cpp]: Tests covered in `test_gemma_norm_3d` and `test_gemma_norm_4d`.

**Review by sunjiweiswift — APPROVED** [review]
